lxd/device/device.go

```diff
@@ -0,0 +1,128 @@
+package device
+
+import (
+	"fmt"
+
+	"github.com/lxc/lxd/lxd/device/config"
+	"github.com/lxc/lxd/lxd/state"
+)
+
+// devTypes defines supported top-level device type creation functions.
+var devTypes = map[string]func(config.Device) device{
+	"nic": loadNICByType,
+}
+
+// ErrUnsupportedDevType is the error that occurs when an unsupported device type is created.
+var ErrUnsupportedDevType = fmt.Errorf("Unsupported device type")
```

Comment:
Maybe move to `errors.go` so when those start piling up they'll be easier to locate?

Reply:
@stgraber makes sense, will move to errors.go